prediction_market_agent/agents/microchain_agent/app.py

```diff
@@ -40,9 +30,48 @@ def run_agent(agent: Agent, iterations: int, model: str) -> None:
 def execute_reasoning(agent: Agent, reasoning: str, model: str) -> None:
     with openai_costs(model) as costs:
         agent.execute_command(f'Reasoning("{reasoning}")')
+        display_new_history_callback(agent)  # Run manually after `execute_command`
         st.session_state.running_cost += costs.cost
 
 
+def display_all_history(agent: Agent) -> None:
+    """
+    Display the agent's history in the Streamlit app.
+    """
+    # Skip the initial messages
+    history = agent.history[get_initial_history_length(agent) :]
+
+    for h in history:
+        st.chat_message(h["role"]).write(h["content"])
+
```

Comment:
I see some errors in the console (e.g. function called with wrong number of args) that don't pop-up in the dashboard.
Maybe we can display those if the user toggles a "Debug" button or similar?

![image](https://github.com/gnosis/prediction-market-agent/assets/4126235/07d6a2ed-20aa-492d-b749-24b8c48da3a1)

Reply:
Yeah that's annoying - I'm not sure how that would work cleanly with microchain. Those statements are `print`ed (e.g. here https://github.com/galatolofederico/microchain/blob/main/microchain/engine/agent.py#L27), not logged or stored in the agent's history.

Reply:
Ow I see - it would be nice to see the "complete" history including errors. Let's try to achieve that on the autogen one!